### PR TITLE
Enable EigenProblem user set vector/matrix tag support

### LINKED ISSUE #15681: Residual vector tags should work with EigenProblem
## Reason
User-defined residual vector tags do not currently work with the EigenProblem because the residual evaluation disassociates all tagged vectors (including the user's tagged vectors).

## Design
Do not disassociate the user's tagged vectors so that user's residual vector tags work with the EigenProblem.

## Impact
Enhanced residual vector tag support

### PR DESCRIPTION
**Pull request by loganharbour**
Also cleans some of the vector/matrix tag interfaces.

Closes #15681

### CODE REVIEW

**Comment by loganharbour** [comment]
Need to get https://github.com/libMesh/libmesh/pull/2638 in so that matrix supports exists for the `NonlinearEigenSystem`. The test is ready to go once matrix support is there: just need to uncomment and regold.

**Comment by fdkong** [comment]
This looks great for me! Big improvement!

**Comment by loganharbour** [comment]
Go away stale bot

File: framework/src/problems/EigenProblem.C [hunk 1]
```diff
@@ -129,13 +129,21 @@ EigenProblem::computeJacobianTag(const NumericVector<Number> & soln,
...
+  // Disassociate the default tags because we will associate vectors with only the
+  // specific system tags that we need for this instance
+  _nl_eigen->disassociateDefaultMatrixTags();
 
+  // Clear FE tags and first add the specific tag assoicated with the Jacobian
+  _fe_matrix_tags.clear();
   _fe_matrix_tags.insert(tag);
 
-  _nl_eigen->setSolution(soln);
+  // Add any other user-added matrix tags if they have associated matrices
```
Comment (fdkong) [inline framework/src/problems/EigenProblem.C:140]: Nice!

**Review by fdkong — APPROVED** [review]
Great Job! Thanks

**Comment by fdkong** [comment]
I can not merge it because some tests are failing. I know the fails are not related to this work, it is better to wait until the tests are clean

**Pull request merged**